src/sagemaker/processing.py

```diff
@@ -1804,7 +1805,7 @@ def _pack_and_upload_code(
             raise RuntimeError("S3 source_dir file must be named `sourcedir.tar.gz.`")
 
         script = estimator.uploaded_code.script_name
-        s3_runproc_sh = self._create_and_upload_runproc(script, kms_key, entrypoint_s3_uri)
+        s3_runproc_sh = self._create_and_upload_runproc(script, self.output_kms_key, entrypoint_s3_uri)
```

Comment:
@raj-oli thanks for the contribution, but what if a custom kms key is set by the method argument `kms_key`, I think we need to set kms_key = self.output_kms_key if kms_key=None

Reply:
@mufaddal-rohawala. Thanks for the heads up. You're right !!.
I'll close this PR as you've already merged the fix. Thank you